staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -566,8 +566,15 @@ func CleanVerb(verb string, request *http.Request) string {
 }
 
 // cleanVerb additionally ensures that unknown verbs don't clog up the metrics.
-func cleanVerb(verb string, request *http.Request) string {
+func cleanVerb(verb, suggestedVerb string, request *http.Request) string {
 	reportedVerb := CleanVerb(verb, request)
+	// CanonicalVerb (being an input for this function) doesn't handle correctly the
+	// deprecated path pattern for watch of:
+	//   GET /api/{version}/watch/{resource}
+	// We correct it manually based on the pass verb from the installer.
+	if suggestedVerb == "WATCH" || suggestedVerb == "WATCHLIST" {
```

Comment:
do we need to check the url directly for the watch segment or is this sufficient?

Reply:
I checked, this is probably okay.